api/src/main/java/marquez/common/models/SourceType.java

```diff
@@ -14,13 +14,42 @@
 
 package marquez.common.models;
 
-public enum SourceType {
-  BIGQUERY,
-  PULSAR,
-  ORACLE,
-  MYSQL,
-  POSTGRESQL,
-  REDSHIFT,
-  SNOWFLAKE,
-  KAFKA;
+import static marquez.common.base.MorePreconditions.checkNotBlank;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.util.StdConverter;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+
+@EqualsAndHashCode
+@ToString
+@JsonDeserialize(converter = SourceType.FromValue.class)
+@JsonSerialize(converter = SourceType.ToValue.class)
+public final class SourceType {
+  @Getter private final String value;
+
+  public SourceType(@NonNull final String value) {
+    this.value = checkNotBlank(value, "value must not be blank").toUpperCase().trim();
```

Comment:
I don't see the benefit from uppercasing their data. Why would we modify the users data? It can be frustrating to work with systems that change your data.

Reply:
My initial thought was to keep types upper case to be consistent with types in other models (`Dataset`, `Job`), but given that this is a user provided field, and not a field that is set / owned by Marquez, makes sense to not modify it.